### PR TITLE
Update documentation and samples to use node.roles

### LINKED ISSUE #3410: Node roles: update documentation
Following up #3387 we should maybe update https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-node-configuration.html
to reflect the fact that individual settings `node.*` are deprecated in favor of a `node.roles` setting in 7.9.

Related to https://github.com/elastic/cloud-on-k8s/issues/3324

### PR DESCRIPTION
**Pull request by charith-elastic**
Uses `node.roles` instead of `node.{role_name}` in the documentation and samples.

Fixes #3410

### CODE REVIEW

**Comment by thbkrkr** [comment]
I guess [`elasticsearch/orchestration.asciidoc`](https://github.com/elastic/cloud-on-k8s/blob/master/docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc) fell through the cracks because we forgot the `node.` prefix there (highlighted with #3699). Maybe it's the occasion to tackle it to not forget it?

File: docs/orchestrating-elastic-stack-applications/elasticsearch/node-configuration.asciidoc [hunk 1]
```diff
@@ -17,22 +17,31 @@ spec:
...
+----
+spec:
+  nodeSets:
+  - name: coordinator
+    count: 3
+    config:
+      node.roles: !!seq ""
+----
+
+====
```
Comment (sebgl) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/node-configuration.asciidoc:47]: Should we add another note to mention that before Elasticsearch 7.9.0, people should use the older roles specification?

```yaml
config:
      node.master: false
      node.data: true
      node.ingest: true
      node.ml: true
      node.transform: true
```

**Review by sebgl — APPROVED** [review]
LGTM.
I hope we eventually find a better solution for the coordinator node :/

**Comment by charith-elastic** [comment]
@elasticmachine run elasticsearch-ci/docs

**Pull request merged**